src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,98 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Comparator;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] data = getSortedData(getSumItems(fromFileName));
+        StringBuilder resultData = new StringBuilder();
+        for (String line : data) {
+            resultData.append(line).append(System.lineSeparator());
+        }
+        resultData.append(getOdd(data));
+        writeToFile(resultData.toString().split(System.lineSeparator()), toFileName);
+    }
+
+    private String[] getSortedData(String[] resultData) {
+        Arrays.sort(resultData, new Comparator<String>() {
+            @Override
+            public int compare(String o1, String o2) {
+                return Integer.valueOf(o2.split(",")[1])
+                        .compareTo(Integer.valueOf(o1.split(",")[1]));
+            }
+        });
+        return resultData;
+    }
+
+    private String[] getSumItems(String fromFileName) {
+        String[] data = readFromFile(fromFileName);
+        StringBuilder resultData = new StringBuilder();
+        String[] keyWords = getUniqueWords(data);
+        int counter = 0;
+        for (String datum : keyWords) {
+            for (String word : data) {
+                String[] compareWords = word.split(",");
+                if (datum.equals(compareWords[0])) {
+                    counter += Integer.parseInt(compareWords[1]);
+                }
+            }
+            resultData.append(datum).append(",").append(counter).append(System.lineSeparator());
```

Comment:
This line is using String concatenation inside a loop. Consider using StringBuilder's append method to improve performance.